sdk/appconfiguration/app-configuration/src/index.ts

```diff
@@ -189,8 +187,11 @@ export class AppConfigurationClient {
    */
   getConfigurationSetting(
     key: string,
-    options?: GetConfigurationSettingOptions
+    options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
+    options = { ...options };
```

Comment:
Why is this step needed?

Reply:
Just below this we're about to set the .label field. So it's only needed so we have a copy and don't modify the user's input parameter.